aten/src/ATen/native/Pool.h

```diff
@@ -68,8 +68,20 @@ pool2d_shape_check(
               "dilation should be greater than zero, but got ",
               "dilationH: ", dilationH, " dilationW: ", dilationW);
 
-  TORCH_CHECK(input.numel() > 0 && (ndim == 3 || ndim == 4),
-              "non-empty 3D or 4D input tensor expected but got ndim: ", ndim);
+  const auto memory_format = input.suggest_memory_format();
+  
+  bool valid_dims = input.size(1) != 0 && input.size(2) != 0;
+  if (memory_format == at::MemoryFormat::ChannelsLast){
+    // Expect tensor in NHWC format and allow 0-dim only for N.
+    TORCH_CHECK((input.ndimension() == 4 && valid_dims && input.size(3) != 0),
+      "4D (batch mode) tensor expected for input with channels_last layout");
```

Comment:
Can we also print `input.sizes()` with the error message?

Reply:
Maybe also reword a bit to clarify the logic, such as what you wrote below, "optional 0 dim batch size for input, but got..."

Reply:
OK will do.